### PR TITLE
cmd/k8s-operator/deploy/chart: allow reading OAuth creds from a CSI driver's volume and annotating operator's Service account

### PR DESCRIPTION
**Pull request by oliverrahner**
This is an alternative approach to #11270 / #12213

It does intentionally _not_ follow the `extraVolumes`/`extraMounts` pattern to increase ease-of-use, but solves the very specific problem of using [secrets-store-csi-driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver).

Not setting the new object results in falling back to the previous default behavior.

### CODE REVIEW

File: cmd/k8s-operator/deploy/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -35,9 +35,13 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       volumes:
-      - name: oauth
-        secret:
-          secretName: operator-oauth
+        - name: oauth
+        {{- with .Values.oauthSecretVolume }}
+        {{- toYaml . | nindent 8 }}
```
Comment (irbekrm) [inline cmd/k8s-operator/deploy/chart/templates/deployment.yaml:40]: I think the indent needs to be 10 here?

File: cmd/k8s-operator/deploy/chart/values.yaml [hunk 2]
```diff
@@ -8,6 +8,17 @@ oauth: {}
   # clientId: ""
   # clientSecret: ""
 
+# Secret volume. If set it defines the volume the oauth secrets will be mounted
```
Comment (irbekrm) [inline cmd/k8s-operator/deploy/chart/values.yaml:11]: Could we also document that:
- the OAuth client creds needs to be provided via `client_id`, `client_secret` keys
- this block would override the `.oauth` block

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thank you for working on this @oliverrahner , this looks neat and makes sense.

I've left two comments, additionally could you please reference an issue number on the commit message to pass our CI bot, see an example https://github.com/tailscale/tailscale/commit/9f9063e624c66d295d286d2f7bc85c02dfd46d4f

File: cmd/k8s-operator/deploy/chart/templates/deployment.yaml (hunk 1 above)
Comment (oliverrahner) [reply cmd/k8s-operator/deploy/chart/templates/deployment.yaml:40]: Good spot... wonder why it works for me the way it is. Will check

Comment (oliverrahner) [reply cmd/k8s-operator/deploy/chart/templates/deployment.yaml:40]: 8 is actually correct. We want to be on the same level as `name`, not below

File: cmd/k8s-operator/deploy/chart/values.yaml (hunk 2 above)
Comment (oliverrahner) [reply cmd/k8s-operator/deploy/chart/values.yaml]: done

File: cmd/k8s-operator/deploy/chart/templates/deployment.yaml (hunk 1 above)
Comment (irbekrm) [reply cmd/k8s-operator/deploy/chart/templates/deployment.yaml:40]: I think the name is indented by 10 (the `-` sign is indented by 8)
It fails for me locally if I just uncomment the new `oauthSecretVolume` block and run `helm template` with
```
Error: YAML parse error on tailscale-operator/templates/deployment.yaml: error converting YAML to JSON: yaml: line 23: did not find expected '-' indicator
```
and succeeds if I change the indent to 10.

Comment (oliverrahner) [reply cmd/k8s-operator/deploy/chart/templates/deployment.yaml]: Found the issue 😅 Had two versions of this, one in my IaC repo and one in the fork of your repo.
The one in my IaC repo looked like this:
```
      volumes:
      - name: oauth
```
Fixed it now, according to the style used throughout your `yaml`s

**Review by irbekrm — APPROVED** [review]
Thank you @oliverrahner !

**Comment by irbekrm** [comment]
@oliverrahner I've cut a dev Helm chart with your change - `1.77.162`. It will also come out in our stable release this or next week.

**Comment by oliverrahner** [comment]
Well, that was one of the quickest PR reviews I ever had :) Thanks a bunch!

**Pull request merged**